livox_ros2_driver/livox_ros2_driver/lddc.cpp

```diff
@@ -718,427 +709,461 @@ void Lddc::onDiagnosticsTimer()
+      }
+
+      updater_.add(
+          fmt::format("livox_service_life_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkServiceLife, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_fan_status_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkFan, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_time_sync{}", broadcast_code),
+          std::bind(
+              &Lddc::checkTimeSync, this, std::placeholders::_1,
+              broadcast_code));
+
+      updater_.add(
+          fmt::format("livox_connection_{}", broadcast_code),
+          std::bind(
+              &Lddc::checkConnection, this, std::placeholders::_1,
+              broadcast_code));
+    }
+  }
+
   updater_.force_update();
 }
 
-void Lddc::checkTemperature(diagnostic_updater::DiagnosticStatusWrapper & stat)
+void Lddc::getLidarByBroadcastcode(std::pair<std::string, DeviceInfo *> &lidar,
+                                   const std::string &broadcast_code)
 {
-  if (lidar_count_ == 0) {
-    stat.summary(DiagStatus::WARN, "No LiDARs Connected");
-    return;
+  for (const auto & tmp_lidar : lds_->connected_lidars_) {
+    const std::string tmp_broadcast_code = tmp_lidar.first;
+    if (tmp_broadcast_code == broadcast_code) lidar = tmp_lidar;
   }
+}
 
-  int whole_level = DiagStatus::OK;
-
-  for (const auto & lidar : lds_->connected_lidars_) {
-    int level = DiagStatus::OK;
+void Lddc::checkTemperature(diagnostic_updater::DiagnosticStatusWrapper &stat,
+                            const std::string & broadcast_code)
+{
+  int level = DiagStatus::OK;
+  std::string error_str = "OK";
 
-    const auto & broadcast_code = lidar.first;
-    const auto & device_info = lidar.second;
+  std::pair<std::string, DeviceInfo*> lidar;
+  getLidarByBroadcastcode(lidar, broadcast_code);
 
-    if (device_info == nullptr) {
-      stat.add(broadcast_code, "disconnected");
-      continue;
-    }
-
-    if (device_info->state == kLidarStateInit) {
-      stat.addf(broadcast_code, "%d%%", device_info->status.progress);
-      continue;
-    }
+  const auto & device_info = lidar.second;
 
+  if (device_info == nullptr)
+  {
+    error_str = "disconnected";
+  }
+  else if (device_info->state == kLidarStateInit)
+  {
+    error_str = fmt::format("progress: %d%%", device_info->status.progress);
+  }
+  else
+  {
     TemperatureStatus status = static_cast<TemperatureStatus>(
-      device_info->status.status_code.lidar_error_code.temp_status);
-    if (status == TemperatureStatus::HighOrLow) {
+        device_info->status.status_code.lidar_error_code.temp_status);
+    if (status == TemperatureStatus::HighOrLow)
+    {
       level = DiagStatus::WARN;
-    } else if (status == TemperatureStatus::ExtremelyHighOrLow) {
+    }
+    else if (status == TemperatureStatus::ExtremelyHighOrLow)
+    {
```

Comment:
unnecessary formatting?

Reply:
which formatting is better?

Reply:
Generally in ROS2, `ament_uncrustify` is better, but if the existing code is not formatted, you should ignore them to avoid non-essential diff in a big PR.

Reply:
thanks. The exisiting code is not formatted and cause big diff, but `if` is like `ament_uncrusity`.

Reply:
I change only the format of if.
https://github.com/tier4/livox_ros2_driver/pull/14/commits/92e80e5dcdebcf054bd7ad6dbbe97cbf02329a1b